protostar-rust/src/lib.rs

```diff
@@ -146,15 +153,29 @@ pub fn run(
     );
 
     let mut tests_stats = TestsStats::default();
-    for tests_from_file in tests {
-        run_tests(tests_from_file, &mut tests_stats)?;
+    let mut tests_iterator = tests.into_iter();
+
+    for tests_from_file in tests_iterator.by_ref() {
+        if run_tests(tests_from_file, &mut tests_stats, runner_config)? == RunnerStatus::TestFailed
+        {
+            break;
+        }
     }
+
+    for tests_from_file in tests_iterator {
+        tests_stats.skipped += tests_from_file.tests_configs.len();
+    }
```

Comment:
I think it's safer that the previous approach with a flag.

@mkaput can you take a look at this and compare it with the version from this commit? https://github.com/software-mansion/protostar/pull/2143/commits/71764779af88b914247968fd8b0bcd6e4106c1a9

Or perhaps you have an idea for alternative solution?

Reply:
it's fine